src/main/java/org/cyclops/integrateddynamics/core/part/panel/PartTypePanelVariableDriven.java

```diff
@@ -144,6 +144,9 @@ public void update(INetwork network, IPartNetwork partNetwork, PartTarget target
             // tick-1: Part tick: update the value again, the old value has still not been sent here!
             // tick-1: -- send all block updates to client --- This will contain the value that was set in tick-1.
             state.onDirty();
+            if (network != null) {
+                network.getEventBus().post(new VariableContentsUpdatedEvent(network));
```

Comment:
Placing this call here will cause panels to trigger `VariableContentsUpdatedEvent` every time the displayed value changes. This means that showing something like the world time will become _very_ expensive.

Reply:
This is the tough one...
I've removed this call and instead added a call in ContainerPartDisplay#onDirty which prevents the performance problem but also detects changes when display inventories are changed.
The remaining problem then is that if you have a display that is displaying the result of an operator that combines the values from two other displays then it does not update.
This screenshot shows a test network I've been using:
![image](https://user-images.githubusercontent.com/17239590/36631182-ce795494-1930-11e8-9aa2-39d11178187e.png)
The bottom row of displays contain redstone cards from the readers below, lets say they have ids 1, 2, 3, 4 in that order.
The second row up has variable cards that are set by 'anding' together the following variables from below (1 && 2, 1 && 2, 3 && 4, 3 && 4), and they have ids 5, 6, 7, 8.  This means that if you switch off either of the two switches on the left then both the left hand lamp and left hand display on the second row should go off and false.  Similar on the right.  Even with removing this call, this row still works fine.
The problem occurs in the third row, where the top two displays are made by (6 && 7, 5 && 8), i.e the left hand display is the two cards from the displays on the second row 'anded' together and the right hand display is the two cards from the writers on the second row 'anded' together.
This means that if any of the switches are off then both top displays should be false, and if all are on then both should be true.
The problem is that the top left display (cards from displays combined in an operator) does not update when I flip one of the switches on the second row.
I'm going to go ahead and push the latest version so we can discuss how to proceed.
If you need a test world with the setup in I can supply that.  I'm also working on getting an integration test setup with the network, but that is going to take time :)
On that note I hope that my attempts here aren't taking up too much of your time -- I'm kind of using this as a way to learn MC modding in general...